modules/economy-system/commands/shop.js

```diff
@@ -1,83 +1,55 @@
-const {createShopItem, editBalance, createShopMsg, deleteShopItem, createleaderboard} = require('../economy-system');
-const {embedType, formatDiscordUserName} = require('../../../src/functions/helpers');
+const {createShopItem, createShopMsg, deleteShopItem, shopMsg, buyShopItem} = require('../economy-system');
 const {localize} = require('../../../src/functions/localize');
 
+/**
+ * @param {*} interaction Interaction
+ * @returns {boolean} Result
+ */
+async function checkPerms(interaction) {
+    const result = interaction.client.configurations['economy-system']['config']['shopManagers'].includes(interaction.user.id) || interaction.client.config['botOperators'].includes(interaction.user.id);
+    if (!result) {
+        await interaction.reply({
+            content: interaction.client.strings['not_enough_permissions'],
+            ephemeral: !interaction.client.configurations['economy-system']['config']['publicCommandReplies']
+        });
+    }
+    return result;
+}
+
 module.exports.subcommands = {
     'add': async function (interaction) {
-        if (!interaction.client.configurations['economy-system']['config']['shopManagers'].includes(interaction.user.id) && !interaction.client.config['botOperators'].includes(interaction.user.id)) return interaction.reply(embedType(interaction.client.strings['not_enough_permissions'], {}, {ephemeral: !interaction.client.configurations['economy-system']['config']['publicCommandReplies']}));
-        const item = await interaction.options.get('item');
-        const price = await interaction.options.getInteger('price');
-        const role = await interaction.options.getRole('role', true);
-        await createShopItem(item['value'], price, role.id, interaction.client);
-        interaction.reply(embedType(interaction.client.configurations['economy-system']['strings']['itemCreate'], {
-            '%item%': item['value'],
-            '%price%': price,
-            '%role%': role.name
-        }, {ephemeral: !interaction.client.configurations['economy-system']['config']['publicCommandReplies']}));
-        interaction.client.logger.info(`[economy-system] ` + localize('economy-system', 'created-item', {
-            u: formatDiscordUserName(interaction.user),
-            i: item['value']
-        }));
-        if (interaction.client.logChannel) interaction.client.logChannel.send(`[economy-system] ` + localize('economy-system', 'created-item', {
-            u: formatDiscordUserName(interaction.user),
-            i: item['value']
-        }));
+        if (!checkPerms(interaction)) return;
+        interaction.reply({
+            content: localize('economy-system', 'creating-item'),
+            ephemeral: true
+        });
```

Comment:
```suggestion
        interaction.reply({
            content: localize('economy-system', 'creating-item'),
            ephemeral: true
        });
```
This creates a bunch of issues. What you are likely looking for is first the `await interaction.deferReply({ephemeral: true})` call, after which all `interaction.reply` calls would need to be replaced by `editReply` calls to avoid issues (same with other affected commands, such as /shop delete`.